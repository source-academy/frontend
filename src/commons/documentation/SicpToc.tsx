import { Tree, TreeNodeInfo } from '@blueprintjs/core';
import { cloneDeep } from 'lodash';
import React, { useState } from 'react';
import toc from 'src/features/sicp/data/toc.json';

type Props = {
  handleCloseToc?: () => void;
  handleNodeClicked?: (node: TreeNodeInfo) => void;
};

/**
 * Table of contents of SICP.
 */
const SicpToc: React.FC<Props> = props => {
  const [sidebarContent, setSidebarContent] = useState(toc as TreeNodeInfo[]);

  const handleNodeExpand = (_node: TreeNodeInfo, path: integer[]) => {
    const newState = cloneDeep(sidebarContent);
    Tree.nodeFromPath(path, newState).isExpanded = true;
    setSidebarContent(newState);
  };

  const handleNodeCollapse = (_node: TreeNodeInfo, path: integer[]) => {
    const newState = cloneDeep(sidebarContent);
    Tree.nodeFromPath(path, newState).isExpanded = false;
    setSidebarContent(newState);
  };

  return (
    <div className="sicp-toc sicp-toc-assessment">
      <Tree
        className="sicp-toc-tree"
        contents={sidebarContent}
        onNodeClick={props.handleNodeClicked}
        onNodeCollapse={handleNodeCollapse}
        onNodeExpand={handleNodeExpand}
      />
    </div>
  );
};

export default SicpToc;
