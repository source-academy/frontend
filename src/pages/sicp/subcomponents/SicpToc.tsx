import { Tree, TreeNodeInfo } from '@blueprintjs/core';
import { cloneDeep } from 'lodash';
import * as React from 'react';
import { useState } from 'react';
import { useHistory } from 'react-router';

import toc from '../../../features/sicp/data/toc.json';

type TocProps = OwnProps;

type OwnProps = {
  handleCloseToc?: () => void;
};

/**
 * Table of contents of SICP.
 */
const SicpToc: React.FC<TocProps> = props => {
  const [sidebarContent, setSidebarContent] = useState(toc as TreeNodeInfo[]);
  const history = useHistory();

  const handleNodeExpand = (_node: TreeNodeInfo, path: integer[]) => {
    const newState = cloneDeep(sidebarContent);
    Tree.nodeFromPath(path, newState).isExpanded = true;
    setSidebarContent(newState);
  };

  const handleNodeCollapse = (_node: TreeNodeInfo, path: integer[]) => {
    const newState = cloneDeep(sidebarContent);
    Tree.nodeFromPath(path, newState).isExpanded = false;
    setSidebarContent(newState);
  };

  const handleNodeClicked = React.useCallback(
    (node: TreeNodeInfo) => {
      if (props.handleCloseToc) {
        props.handleCloseToc();
      }
      history.push('/sicpjs/' + String(node.nodeData));
    },
    [history, props]
  );

  return (
    <div className="sicp-toc">
      <Tree
        className="sicp-toc-tree"
        contents={sidebarContent}
        onNodeClick={handleNodeClicked}
        onNodeCollapse={handleNodeCollapse}
        onNodeExpand={handleNodeExpand}
      />
    </div>
  );
};

export default SicpToc;
