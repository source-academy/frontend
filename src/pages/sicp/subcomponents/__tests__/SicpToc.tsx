import { mount } from 'enzyme';

import SicpToc from '../SicpToc';

test('Sicp toc renders correctly', () => {
  const props = {
    handleCloseToc: () => {}
  };

  const tree = mount(<SicpToc {...props} />);
  expect(tree.debug()).toMatchSnapshot();
});
