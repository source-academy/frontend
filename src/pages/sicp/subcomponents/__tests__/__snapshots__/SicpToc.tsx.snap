// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sicp toc renders correctly 1`] = `
"<SicpToc handleCloseToc={[Function: handleCloseToc]}>
  <div className=\\"sicp-toc\\">
    <Blueprint4.Tree className=\\"sicp-toc-tree\\" contents={{...}} onNodeClick={[Function (anonymous)]} onNodeCollapse={[Function: handleNodeCollapse]} onNodeExpand={[Function: handleNodeExpand]}>
      <div className=\\"bp4-tree sicp-toc-tree\\">
        <ul className=\\"bp4-tree-node-list bp4-tree-root\\">
          <Blueprint4.TreeNode id={0} hasCaret={false} label=\\"Foreword\\" nodeData=\\"foreword02\\" contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <span className=\\"bp4-tree-node-caret-none\\" />
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  Foreword
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={1} hasCaret={false} label=\\"Foreword to Structure and Interpretation of Computer Programs, 1984\\" nodeData=\\"foreword84\\" contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <span className=\\"bp4-tree-node-caret-none\\" />
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  Foreword to Structure and Interpretation of Computer Programs, 1984
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={2} hasCaret={false} label=\\"Preface\\" nodeData=\\"prefaces03\\" contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <span className=\\"bp4-tree-node-caret-none\\" />
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  Preface
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={3} hasCaret={false} label=\\"Prefaces to Structure and Interpretation of Computer Programs, 1996 & 1984\\" nodeData=\\"prefaces96\\" contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <span className=\\"bp4-tree-node-caret-none\\" />
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  Prefaces to Structure and Interpretation of Computer Programs, 1996 &amp; 1984
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={4} hasCaret={false} label=\\"Acknowledgments\\" nodeData=\\"acknowledgements\\" contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <span className=\\"bp4-tree-node-caret-none\\" />
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  Acknowledgments
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={6} hasCaret={true} label=\\"1 Building Abstractions with Functions\\" nodeData=\\"1\\" childNodes={{...}} contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <Blueprint4.Icon title=\\"Expand group\\" className=\\"bp4-tree-node-caret bp4-tree-node-caret-closed\\" onClick={[Function (anonymous)]} icon=\\"chevron-right\\">
                  <span aria-hidden={[undefined]} onClick={[Function (anonymous)]} icon=\\"chevron-right\\" className=\\"bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed\\" title={[undefined]}>
                    <svg fill={[undefined]} data-icon=\\"chevron-right\\" width={16} height={16} viewBox=\\"0 0 16 16\\" aria-labelledby=\\"iconTitle-0\\" role=\\"img\\">
                      <title id=\\"iconTitle-0\\">
                        Expand group
                      </title>
                      <path d=\\"M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z\\" fillRule=\\"evenodd\\" />
                    </svg>
                  </span>
                </Blueprint4.Icon>
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  1 Building Abstractions with Functions
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={28} hasCaret={true} label=\\"2 Building Abstractions with Data\\" nodeData=\\"2\\" childNodes={{...}} contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <Blueprint4.Icon title=\\"Expand group\\" className=\\"bp4-tree-node-caret bp4-tree-node-caret-closed\\" onClick={[Function (anonymous)]} icon=\\"chevron-right\\">
                  <span aria-hidden={[undefined]} onClick={[Function (anonymous)]} icon=\\"chevron-right\\" className=\\"bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed\\" title={[undefined]}>
                    <svg fill={[undefined]} data-icon=\\"chevron-right\\" width={16} height={16} viewBox=\\"0 0 16 16\\" aria-labelledby=\\"iconTitle-1\\" role=\\"img\\">
                      <title id=\\"iconTitle-1\\">
                        Expand group
                      </title>
                      <path d=\\"M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z\\" fillRule=\\"evenodd\\" />
                    </svg>
                  </span>
                </Blueprint4.Icon>
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  2 Building Abstractions with Data
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={52} hasCaret={true} label=\\"3 Modularity, Objects, and State\\" nodeData=\\"3\\" childNodes={{...}} contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <Blueprint4.Icon title=\\"Expand group\\" className=\\"bp4-tree-node-caret bp4-tree-node-caret-closed\\" onClick={[Function (anonymous)]} icon=\\"chevron-right\\">
                  <span aria-hidden={[undefined]} onClick={[Function (anonymous)]} icon=\\"chevron-right\\" className=\\"bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed\\" title={[undefined]}>
                    <svg fill={[undefined]} data-icon=\\"chevron-right\\" width={16} height={16} viewBox=\\"0 0 16 16\\" aria-labelledby=\\"iconTitle-2\\" role=\\"img\\">
                      <title id=\\"iconTitle-2\\">
                        Expand group
                      </title>
                      <path d=\\"M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z\\" fillRule=\\"evenodd\\" />
                    </svg>
                  </span>
                </Blueprint4.Icon>
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  3 Modularity, Objects, and State
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={77} hasCaret={true} label=\\"4 Metalinguistic Abstraction\\" nodeData=\\"4\\" childNodes={{...}} contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <Blueprint4.Icon title=\\"Expand group\\" className=\\"bp4-tree-node-caret bp4-tree-node-caret-closed\\" onClick={[Function (anonymous)]} icon=\\"chevron-right\\">
                  <span aria-hidden={[undefined]} onClick={[Function (anonymous)]} icon=\\"chevron-right\\" className=\\"bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed\\" title={[undefined]}>
                    <svg fill={[undefined]} data-icon=\\"chevron-right\\" width={16} height={16} viewBox=\\"0 0 16 16\\" aria-labelledby=\\"iconTitle-3\\" role=\\"img\\">
                      <title id=\\"iconTitle-3\\">
                        Expand group
                      </title>
                      <path d=\\"M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z\\" fillRule=\\"evenodd\\" />
                    </svg>
                  </span>
                </Blueprint4.Icon>
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  4 Metalinguistic Abstraction
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={99} hasCaret={true} label=\\"5 Computing with Register Machines\\" nodeData=\\"5\\" childNodes={{...}} contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <Blueprint4.Icon title=\\"Expand group\\" className=\\"bp4-tree-node-caret bp4-tree-node-caret-closed\\" onClick={[Function (anonymous)]} icon=\\"chevron-right\\">
                  <span aria-hidden={[undefined]} onClick={[Function (anonymous)]} icon=\\"chevron-right\\" className=\\"bp4-icon bp4-icon-chevron-right bp4-tree-node-caret bp4-tree-node-caret-closed\\" title={[undefined]}>
                    <svg fill={[undefined]} data-icon=\\"chevron-right\\" width={16} height={16} viewBox=\\"0 0 16 16\\" aria-labelledby=\\"iconTitle-4\\" role=\\"img\\">
                      <title id=\\"iconTitle-4\\">
                        Expand group
                      </title>
                      <path d=\\"M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z\\" fillRule=\\"evenodd\\" />
                    </svg>
                  </span>
                </Blueprint4.Icon>
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  5 Computing with Register Machines
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={127} hasCaret={false} label=\\"References\\" nodeData=\\"references\\" contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <span className=\\"bp4-tree-node-caret-none\\" />
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  References
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
          <Blueprint4.TreeNode id={128} hasCaret={false} label=\\"About the SICP JS Project\\" nodeData=\\"making-of\\" contentRef={[Function (anonymous)]} depth={0} onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onCollapse={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onExpand={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]} path={{...}}>
            <li className=\\"bp4-tree-node\\">
              <div className=\\"bp4-tree-node-content bp4-tree-node-content-0\\" onClick={[Function (anonymous)]} onContextMenu={[Function (anonymous)]} onDoubleClick={[Function (anonymous)]} onMouseEnter={[Function (anonymous)]} onMouseLeave={[Function (anonymous)]}>
                <span className=\\"bp4-tree-node-caret-none\\" />
                <Blueprint4.Icon className=\\"bp4-tree-node-icon\\" icon={[undefined]} aria-hidden={true} tabIndex={-1} />
                <span className=\\"bp4-tree-node-label\\">
                  About the SICP JS Project
                </span>
              </div>
              <Blueprint4.Collapse isOpen={false} component=\\"div\\" keepChildrenMounted={false} transitionDuration={200}>
                <div className=\\"bp4-collapse\\" style={{...}}>
                  <div className=\\"bp4-collapse-body\\" style={{...}} aria-hidden={true} />
                </div>
              </Blueprint4.Collapse>
            </li>
          </Blueprint4.TreeNode>
        </ul>
      </div>
    </Blueprint4.Tree>
  </div>
</SicpToc>"
`;
